New Quickstart for golem-js@3 (#278)
golemfactory/golem-docs: src/pages/docs/creators/javascript/guides/golem-js.md

- define a method that will select which proposal should be chosen first. This is useful when you want to prioritize certain providers over others,
- avoid uploading a GVMI image directly to the provider instead of using the `registry` service. This is useful when you want to quickly prototype your image without having to update the registry with every change,
- define a setup method that will be executed the first time a provider is rented and a teardown method that will be executed before the rental is done. This is useful when you want to avoid doing the same work multiple times when running multiple tasks on the same provider.

`Golem-js` also exposes low-level api modules representing Yagna domains in for of glm.market, glm.activity, glm.payment properties of the GolemNetwork object.

{% docnavigation title="Next steps" %}

The [examples](https://github.com/golemfactory/golem-js/tree/master/examples) directory in the repository contains various usage patterns for the SDK. You can browse through them and learn about the recommended practices. All examples are automatically tested during our release process.

> You can also start exploting [API reference](https://docs.golem.network/docs/golem-js/reference/overview).

Reviewer: ```suggestion
You can also start exploring [API reference](https://docs.golem.network/docs/golem-js/reference/overview).
```

{% /docnavigation %}

{% docnavigation title="See also" %}

> If you wish to learn more about how the SDK functions under the hood, please check out our more [advanced examples](https://github.com/golemfactory/golem-js/tree/beta/examples/advanced)

Reviewer: ```suggestion
If you wish to learn more about how the SDK functions under the hood, please check out our more [basic and advanced examples](https://github.com/golemfactory/golem-js/tree/master/examples/rental-model).
```

> If you wish to learn more about the usage for low-level features of the SDK, please check out our [low level API examples](https://github.com/golemfactory/golem-js/tree/beta/examples/advanced)

Reviewer: ```suggestion
If you wish to learn more about the usage for low-level features of the SDK, please check out our [Core API examples](https://github.com/golemfactory/golem-js/tree/master/examples/core-api)
```

> <!-- to do: fix the link for low-level -->

Reviewer: ```suggestion
```

---
description: Introduction to Resource Rental Model
title: Introduction to Resource Rental Model
type: Guide
---

# Introduction to Golem Network and Introduction to Resource Rental Model

## Introduction

This article will introduce you to the Resource Rental model leveraged by the `golem-js` library.

## Golem Network

Golem connects computers in a peer-to-peer network, enabling both application owners and individual users **requestor** to rent resources of other users’ **providers** machines. These resources can be used to complete tasks requiring any amount of computation time and capacity.

> To rent resources you need to create an 'order`that describes your needs then rent the resources from the network using the Yagna service. This is done using a **requestor script** (or simply **Golem app**) that will utilize Golem JS SDK. When you define the`order` you define the software environment in which you will run your commands on a provider. This is done by specifying a software package in the form of a Docker-like **image**.

Author: ```suggestion
To rent resources you need to create an `order` that describes your needs then rent the resources from the network using the Yagna service. This is done using a **requestor script** (or simply **Golem app**) that will utilize Golem JS SDK. When you define the `order` you define the software environment in which you will run your commands on a provider. This is done by specifying a software package in the form of a Docker-like **image**.
```

So, to run your app on Golem Network, you need:

- connection to a Yagna service that will let you interact with the Golem Network.
- A Docker-like [Image](/docs/creators/javascript/guides/golem-images), that defines an environment where you will run your commands.
- A [requestor script](/docs/creators/javascript/tutorials/quickstart-explained), in which you orchestrate the whole process and define commands run on rented resources.

As the script will use API and model provided by the JS SDK `golem-js` library, let’s get familiar with the Resource Rental model.

## Resource Rental Model

Since version 3.0, golem-js leverages the notion of renting compute resources and accordingly exposes domain APIs around it.

> The `GolemNetwork` is the point of entry that represents a connection to the Yagna service and thus the Golem Network. It lets you rent resources and (if you need to implement more customized logic) get access to the core modules (market, activity, payment, rental).

Author: network module missed

- connect your node and multiple providers to a VPN network, This enables communication between the nodes.
- listen to various events that are emitted by the SDK. This enables reaction to certain conditions, like calculating the total cost of all invoices received,
- use a predefined filter for common use cases, to select more suitable providers,
- define custom filters to select the providers that he wants to work with. This enables you to blacklist or whitelist certain providers,
- define a method that will select which proposal should be chosen first. This is useful when you want to prioritize certain providers over others,
- avoid uploading a GVMI image directly to the provider instead of using the `registry` service. This is useful when you want to quickly prototype your image without having to update the registry with every change,
- define a setup method that will be executed the first time a provider is rented and a teardown method that will be executed before the rental is done. This is useful when you want to avoid doing the same work multiple times when running multiple tasks on the same provider.

> `Golem-js` also exposes low-level api modules representing Yagna bdomains in for of glm.market, glm.activity, glm.payment properties of the GolemNetwork object.

Author: ```suggestion
`Golem-js` also exposes low-level api modules representing Yagna domains in for of glm.market, glm.activity, glm.payment properties of the GolemNetwork object.
```

{% docnavigation title="Next steps" %}

> <!-- to define next steps -->

Reviewer: ```suggestion
```

> <!-- to define next steps -->

Reviewer: ```suggestion

```

> The [examples](https://github.com/golemfactory/golem-js/tree/beta/examples) directory in the repository contains various usage patterns for the SDK. You can browse through them and learn about the recommended practices. All examples are automatically tested during our release process.

Reviewer: ```suggestion
The [examples](https://github.com/golemfactory/golem-js/tree/master/examples) directory in the repository contains various usage patterns for the SDK. You can browse through them and learn about the recommended practices. All examples are automatically tested during our release process.

```

> You can also start exploting [API reference]().

Reviewer: ```suggestion
You can also start exploting [API reference](https://docs.golem.network/docs/golem-js/reference/overview).
```
